pkgs/by-name/my/mycelium/package.nix

```diff
@@ -0,0 +1,38 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, stdenv
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "mycelium";
+  version = "0.3.2";
+
+  src = fetchFromGitHub {
+    owner = "threefoldtech";
+    repo = "mycelium";
+    rev = "v${version}";
+    hash = "sha256-eRyjpIloy8zOxP/BDP1PsL82Y7SeSs09uUqVupkH6ro=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "tun-0.6.1" = "sha256-DelNPCOWvVSMS2BNGA2Gw/Mn9c7RdFNR21/jo1xf+xk=";
+    };
+  };
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
+  meta = with lib; {
+    description = "";
+    homepage = "https://github.com/threefoldtech/mycelium";
+    changelog = "https://github.com/threefoldtech/mycelium/blob/${src.rev}/CHANGELOG.md";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ matthewcroughan ];
+    mainProgram = "mycelium";
```

Comment:
What platforms did you try with? At least an impure `cargo build` on aarch64 did not work for me, see https://github.com/threefoldtech/mycelium/issues/132.

Reply:
@flokli only on x86_64-linux, on aarch64-linu I get the following:

```
mycelium>    Compiling futures-channel v0.3.30
mycelium> error: linker `aarch64-linux-gnu-gcc` not found
mycelium>   |
mycelium>   = note: No such file or directory (os error 2)
mycelium> error: could not compile `proc-macro2` (build script) due to previous error
mycelium> warning: build failed, waiting for other jobs to finish...
mycelium> error: could not compile `serde` (build script) due to previous error
mycelium> error: could not compile `thiserror` (build script) due to previous error
mycelium> error: could not compile `libc` (build script) due to previous error
mycelium> error: could not compile `libc` (build script) due to previous error
mycelium> error: could not compile `network-interface` (build s
cript) due to previous error
mycelium> error: could not compile `anyhow` (build script) due to previous error
mycelium> error: could not compile `serde_json` (build script) due to previous error
mycelium> error: could not compile `async-trait` (build script) due to previous error
mycelium> error: could not compile `rustls` (build script) due to previous error
mycelium> error: could not compile `semver` (build script) due to previous error
mycelium> error: could not compile `rustix` (build script) due to previous error
mycelium> error: could not compile `httparse` (build script) due to previous error
mycelium> error: could not compile `rustversion` (build script) due to previous error
mycelium> error: could not compile `typenum` (build script) due to previous error
mycelium> error: could not compile `paste` (build script) due to previous error
```

Reply:
Yeah, that's expected - reported in https://github.com/threefoldtech/mycelium/issues/132 and fixed in https://github.com/threefoldtech/mycelium/commit/cce1cc6c209b7b41ad4b481da28d5d42dee84c06.

With this, I can even cross-compile a `mycelium.exe` - though I don't know if it works or not :laughing:

Reply:
Can you add x86_64-linux? That way we should get binaries built by hydra:

```
### `platforms` {#var-meta-platforms}

The list of Nix platform types on which the package is supported. Hydra builds packages according to the platform specified. If no platform is specified, the package does not have prebuilt binaries. An example is:
```

Reply:
Will wait until 0.4.1

Reply:
Please add `platforms = [ "aarch64-darwin" "aarch64-linux" "x86_64-linux" ];`.